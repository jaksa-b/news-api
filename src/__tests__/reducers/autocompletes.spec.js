import autocompletesReducer from "../../redux/reducers/autocompletes";
import {
  FETCH_AUTOCOMPLETES_REQUEST,
  FETCH_AUTOCOMPLETES_SUCCESS,
  FETCH_AUTOCOMPLETES_FAILED,
} from "../../redux/actions/autocompletes";

describe("Autocompletes", () => {
  it("should return the initial state", () => {
    expect(autocompletesReducer(undefined, {})).toMatchSnapshot();
  });

  it(`should handle ${FETCH_AUTOCOMPLETES_REQUEST}`, () => {
    expect(
      autocompletesReducer(undefined, {
        type: FETCH_AUTOCOMPLETES_REQUEST,
        payload: { term: "apple" },
      })
    ).toMatchSnapshot();
  });

  it(`should handle ${FETCH_AUTOCOMPLETES_SUCCESS}`, () => {
    expect(
      autocompletesReducer(undefined, {
        type: FETCH_AUTOCOMPLETES_SUCCESS,
        payload: {},
      })
    ).toMatchSnapshot();
  });

  it(`should handle ${FETCH_AUTOCOMPLETES_FAILED}`, () => {
    expect(
      autocompletesReducer(undefined, {
        type: FETCH_AUTOCOMPLETES_FAILED,
        payload: {},
      })
    ).toMatchSnapshot();
  });
});
