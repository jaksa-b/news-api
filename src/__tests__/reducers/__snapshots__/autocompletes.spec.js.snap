// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Autocompletes should handle FETCH_AUTOCOMPLETES_FAILED 1`] = `
Object {
  "data": Array [],
}
`;

exports[`Autocompletes should handle FETCH_AUTOCOMPLETES_REQUEST 1`] = `
Object {
  "data": Array [],
}
`;

exports[`Autocompletes should handle FETCH_AUTOCOMPLETES_SUCCESS 1`] = `
Object {
  "data": Array [],
}
`;

exports[`Autocompletes should return the initial state 1`] = `
Object {
  "data": Array [],
}
`;
